app/src/main/java/com/example/hellobuttons/MainActivity.kt

```diff
@@ -27,4 +32,15 @@ class MainActivity : AppCompatActivity() {
             setText(counter.toString())
         }
     }
+
+    // take counter and pass to new activity
+    fun showCounterActivity(view: View) {
+        Log.i("show1", "button pressed")
+
+        // pass
+        val intent = Intent(this, CounterActivity::class.java).apply {
+            putExtra(COUNTER_VALUE, counter)
```

Comment:
I know you mentioned bundle in our chat. I saw some old code creating a bundle when you pass, but that didn't seem to be the way the recent docs do it. is the rule of thumb, you use a bundle if you have a bunch of extras? otherwise, add individually?

Reply:
What were the new docs showing? There is no real other way of passing data between two activities unless you create some sort of session class that stores variables. The latter is fine and how we handle most data at clients. However, the latter doesn't eliminate the need for the former. You will still have use cases of storing data in a bundle and passing it around.

Reply:
new docs show this (putExtra) so I used it, but saw something else where they actually created a bundle of Bundle class then called putInt on the bundle itself. I guess my question is if you have 3 Ints for example you wanted to pass, would you have 3 lines of putExtra? Does that create a "bundle" or do you need to create an actual Bundle object?

Reply:
put extra is another just helper function. It creates the bundle under the hood and adds it for you :)

Reply:
Forgot the other part - no this doesn't create multiple bundles, just one. It essentially checks if the bundle is there and then creates one if not and adds yoru args.

Reply:
thanks! sorry if these are stupid questions

Reply:
Not stupid at all. Good questions!